mitiq/shadows/shadows.py

```diff
@@ -0,0 +1,198 @@
+    executor: Optional[Callable[[cirq.Circuit], MeasurementResult]] = None,
+    num_total_measurements_calibration: Optional[int] = 20000,
+) -> Dict[str, complex]:
+    r"""
+    This function returns the dictionary of the median of means estimation
+    of Pauli fidelities: {:math:`\{'b':f_{b}\}_{b\in\{0,1\}^n}`}.
+
+    Args:
+        k_calibration: Number of groups of "median of means" used for
+            calibration.
+        locality: The locality of the operator, whose expectation value is
+            going to be estimated by the classical shadow. e.g. if operator is
+            Ising model Hamiltonian with nearist neighbour interacting, then
+            locality = 2.
+        zero_state_shadow_outcomes: The output of function
+            `shadow_quantum_processing` of zero calibrate state.
+        qubits: The qubits to measure, needs to spercify when the
+            `zero_state_shadow_outcomes` is None.
+        executor: The function to use to do quantum measurement, must be same
+            as executor in `shadow_quantum_processing`. Needs to spercify when
+            the `zero_state_shadow_outcomes` is None.
+        num_total_measurements_calibration: Number of shots per group of
+            "median of means" used for calibration. Needs to spercify when
+            the `zero_state_shadow_outcomes` is None.
+
+    Returns:
+        A dictionary containing the calibration outcomes.
+    """
+    if zero_state_shadow_outcomes is None:
+        if qubits is None:
+            raise ValueError("qubits should not be None.")
+        if executor is None:
+            raise ValueError("executor should not be None.")
+        if num_total_measurements_calibration is None:
+            raise ValueError(
+                "num_total_measurements_calibration should not be None."
+            )
+
+        # calibration circuit is of same qubit number with original circuit
+        zero_circuit = cirq.Circuit()
+        # perform random Pauli measurement one the calibration circuit
+        calibration_measurement_outcomes = random_pauli_measurement(
+            zero_circuit,
+            n_total_measurements=num_total_measurements_calibration,
+            executor=executor,
+            qubits=qubits,
+        )
+    else:
+        calibration_measurement_outcomes = zero_state_shadow_outcomes
+    # get the median of means estimation of Pauli fidelities
+    return get_pauli_fidelities(
+        calibration_measurement_outcomes, k_calibration, locality=locality
+    )
+
+
+def shadow_quantum_processing(
+    circuit: cirq.Circuit,
+    executor: Callable[[cirq.Circuit], MeasurementResult],
+    num_total_measurements_shadow: int,
+    random_seed: Optional[int] = None,
+    qubits: Optional[List[cirq.Qid]] = None,
+) -> Tuple[List[str], List[str]]:
+    r"""
+    Executes a circuit with classical shadows. This function can be used for
+    state reconstruction or expectation value estimation of observables.
```

Comment:
This description for `shadow_quantum_processing` is the same as for `classical_post_processing`. I don't think that's intended, and I don't think this function is actually executing the classical shadows technique.

Reply:
Thanks for checking this. I have updated the doc string.